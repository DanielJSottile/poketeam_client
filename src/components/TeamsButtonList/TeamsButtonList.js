import React, { Component , Fragment} from 'react';
import UserContext from '../../contexts/UserContext';
import TeamButton from '../TeamButton/TeamButton';

export default class TeamsButtonList extends Component {

  static contextType = UserContext;

  handleOnClick = (e) => {

  }

  handleOnSubmit = (e) => {
    e.preventDefault();
    console.log("hey you clicked me");
    const {team_name} = e.target;

    // some kind of APIservice.PostTeam(team_name.value)

  }

  renderExpanded() {
    return (
      <form onSubmit={this.handleOnSubmit}>
        <div class="team-name">
          <label for="foldername">Team Name:</label>
          <input placeholder="e.g. My Cool Team" type="text" name="teamname" id="teamname" />
        </div>
        <div class="team-import">
          <label for="team-import">Import Team Set:</label>
          <textarea type="text" placeholder="Optionally Import a proper Pokemon Showdown Team Here And It Will Fill Out Your Whole Team!" name="team-import" id="team-import-1"></textarea>
        </div>
        <button type="submit">Submit</button>
      </form>
    )
  }

  render() {

    const {teams} = this.context;

    const TeamList = teams.map((team, i) => {
      return <TeamButton key={i} id={team.id} folder_name={team.team_name}/>
    });

    return (
      <Fragment>
        <section className="folders-list">
          <h3>Teams:</h3>
          <div>
            {TeamList.length > 0 ? TeamList : <h3>None! Click Below to Make a New Team!</h3>}
          </div>
          <div>
            <button onClick={this.handleOnClick}>New Team +</button>
            {clicked ? this.renderExpanded() : null}
          </div>
        <div>
          <span>Current Folder: {currentClickedTeam}</span>
        </div>
      </section>
    </Fragment>
    );
  };
};
