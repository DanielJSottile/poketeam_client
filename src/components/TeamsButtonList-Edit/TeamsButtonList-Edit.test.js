import React from 'react';
import ReactDOM from 'react-dom';
//import { render } from '@testing-library/react';
import TeamsButtonList from './TeamsButtonList-Edit';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<TeamsButtonListEdit/>', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<TeamsButtonList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});