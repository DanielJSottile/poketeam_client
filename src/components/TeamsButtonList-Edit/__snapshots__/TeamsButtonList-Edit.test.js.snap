// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamsButtonListEdit/> renders without crashing 1`] = `
<TeamsButtonList>
  <section
    className="folders-list"
  >
    <h3>
      Teams:
    </h3>
    <div>
      <h3>
        None! Click the Button Below to Get Started!
      </h3>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        New Team 
        <i
          className="far fa-plus-square"
        />
      </button>
    </div>
    <div>
      <span>
        Current Team: undefined
      </span>
    </div>
  </section>
</TeamsButtonList>
`;
